packages/nest/src/feature.decorator.ts

```diff
@@ -17,46 +19,79 @@ interface FeatureProps<T extends FlagValue> {
   contextFactory?: (executionContext: ExecutionContext) => EvaluationContext | undefined;
 }
 
+/**
+ *
+ * @param clientName
+ * @param context
+ */
 function getClientForEvaluation(clientName?: string, context?: EvaluationContext) {
+  //is context thread safe? setting it here might result in context from different requests mixing up
```

Comment:
Yes, getting a named client creates a new instance.
If a provider is bound to the name, all clients with this name get it, but the context is per-instance.